engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/resource/WithNode.java

```diff
@@ -26,19 +25,22 @@ public abstract class WithNode extends Node {
           InvokeCallableNode.ArgumentsExecutionMode.PRE_EXECUTED);
 
   static WithNode build() {
-    return WithNodeGen.create();
+    return new WithNode();
   }
 
-  abstract Object execute(State state, VirtualFrame frame, Object self, Object action);
-
-  @Specialization
-  Object doWith(State state, VirtualFrame frame, ManagedResource self, Object action) {
-    ResourceManager resourceManager = EnsoContext.get(this).getResourceManager();
-    resourceManager.park(self);
-    try {
-      return invokeCallableNode.execute(action, frame, state, new Object[] {self.getResource()});
-    } finally {
-      resourceManager.unpark(self);
+  Object execute(
+      State state, VirtualFrame frame, ManagedResource mr, Object action, Object onMissing) {
+    var ctx = EnsoContext.get(this);
+    var resourceManager = ctx.getResourceManager();
+    if (mr.getPhantomReference().refersTo(mr)) {
+      resourceManager.park(mr);
+      try {
+        return invokeCallableNode.execute(action, frame, state, new Object[] {mr.getResource()});
+      } finally {
+        resourceManager.unpark(mr);
+      }
+    } else {
+      return onMissing;
```

Comment:
According to the docs of `Managed_Resource.with`, `on_missing` is "an optional action", not a constant. Clarify and/or add a test. If it is an action, it should be suspended thunk and called appropriatelly. Maybe even add `@Suspended` annotation to the parameter?

Reply:
`on_missing` removed in 873bac3

Reply:
Why not a panic though? It is not a user-facing error (we want these to be dataflow errors as they play better in the GUI). Any time this happens and is not caught indicates a programming error within some library. So it makes sense to panic hard and make it clear what failed. As I was mentioning several times already, a dataflow error may get unnoticed if it's discarded.